addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,238 @@
+                    "Niko Home Control: cannot resolve bridge IP with hostname " + config.get(CONFIG_HOST_NAME));
+            return;
+        }
+
+        Integer refreshInterval = ((Number) config.get(CONFIG_REFRESH)).intValue();
+
+        createCommunicationObject();
+        setupRefreshTimer(refreshInterval);
+
+    }
+
+    /**
+     * Create communication object to Niko Home Control gateway and start communication.
+     * Trigger discovery when communication setup is successful.
+     *
+     * @param nhcAddr IP address or subnet for gateway
+     * @param port
+     */
+    private void createCommunicationObject() {
+
+        scheduler.submit(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    for (int i = 0; i < 3; i++) {
+                        // try connecting max 3 times
+                        nhcComm = new NikoHomeControlCommunication(addr, port);
+                        if (nhcComm.communicationActive()) {
+                            break;
+                        }
+                        if (i == 2) {
+                            throw new IOException("Niko Home Control: communication socket error");
+                        }
+                    }
+
+                    addr = nhcComm.getAddr();
+                    // keep the discovered host ip in the configuration to reuse when restarting
+                    thing.getConfiguration().put(CONFIG_HOST_NAME, addr.getHostAddress());
+
+                    updateProperties();
+
+                    updateStatus(ThingStatus.ONLINE);
+
+                    if (nhcDiscovery != null) {
+                        nhcDiscovery.discoverDevices();
+                    } else {
+                        logger.debug("Niko Home Control: cannot discover, discovery service not started");
+                    }
+
+                } catch (IOException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                            "Niko Home Control: error starting bridge connection");
+                }
+            }
+        });
+
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+    private void setupRefreshTimer(Integer refreshInterval) {
+
+        if (refreshTimer != null) {
+            refreshTimer.cancel(true);
+            refreshTimer = null;
+        }
+
+        if ((refreshInterval == null) || (refreshInterval == 0)) {
+            return;
+        }
+
+        // This timer will restart the bridge connection periodically
+        logger.debug("Niko Home Control: restart bridge connection every {} min", refreshInterval);
+        refreshTimer = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+
+                logger.debug("Niko Home Control: restart communication at scheduled time");
```

Comment:
I ran into issues when just checking for the connection to still be up. It looks like it was disconnecting automatically at the Niko controller side after a while whatever I did. So I fear that just checking for the connection and not doing anything will get it disconnected after a while. OH will then stop receiving channel updates without knowing anything is wrong until it sends something. I got over it by this periodic restart. I made it configurable. So it could be shortened or switched off if someone sees a different behaviour than I have.